application/comit_node/src/swap_protocols/rfc003/alice/spawner.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{
+    comit_client,
+    swap_protocols::{
+        asset::Asset,
+        dependencies::{LedgerEventDependencies, ProtocolDependencies},
+        metadata_store::{self, Metadata, MetadataStore},
+        rfc003::{
+            alice::SwapRequest,
+            state_store::{self, StateStore},
+            Alice, CreateLedgerEvents, Initiation, Ledger, SecretSource,
+        },
+        SwapId,
+    },
+};
+
+use futures::Future;
+use http_api_problem::HttpApiProblem;
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub enum Error {
+    Storage(state_store::Error),
+    Metadata(metadata_store::Error),
+}
+
+impl From<Error> for HttpApiProblem {
+    fn from(e: Error) -> Self {
+        use self::Error::*;
+        match e {
+            Storage(e) => e.into(),
+            Metadata(e) => e.into(),
+        }
+    }
+}
+
+pub trait AliceSpawner: Send + Sync + 'static {
+    fn spawn_alice<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset>(
+        &self,
+        id: SwapId,
+        swap_request: SwapRequest<AL, BL, AA, BA>,
+    ) -> Result<(), Error>
+    where
+        LedgerEventDependencies: CreateLedgerEvents<AL, AA> + CreateLedgerEvents<BL, BA>,
+        SwapRequest<AL, BL, AA, BA>: Into<Metadata>;
+}
+
+impl<T: MetadataStore<SwapId>, S: StateStore<SwapId>, C: comit_client::Client> AliceSpawner
+    for ProtocolDependencies<T, S, C>
+{
+    fn spawn_alice<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset>(
+        &self,
+        id: SwapId,
+        swap_request: SwapRequest<AL, BL, AA, BA>,
+    ) -> Result<(), Error>
+    where
+        LedgerEventDependencies: CreateLedgerEvents<AL, AA> + CreateLedgerEvents<BL, BA>,
+        SwapRequest<AL, BL, AA, BA>: Into<Metadata>,
+    {
+        let save_state = self
+            .state_store
+            .new_save_state(id)
+            .map_err(Error::Storage)?;
+        self.metadata_store
+            .insert(id, swap_request.clone())
+            .map_err(Error::Metadata)?;
+
+        let initiation = Initiation {
```

Comment:
:upside_down_face: I don't like this struct name.

Reply:
@D4nte @bonomat @thomaseizinger Any ideas on better names? It represents the data needed to start the protocol (as opposed to the services needed).

Reply:
Isn't this almost identical to `OngoingSwap`?

Reply:
@bonomat it's more like the start state.